### ISSUE TITLE
Input Validation example

### ISSUE THREAD

**Issue by PhilippS93** [graphql-java-kickstart/graphql-spring-boot#341]
Hello,

Thank you for your work on this library. This is very helpful.

I want to use a directive to validate input fields like so:

input UserInput {
  name: String @Size( min : 2, max : 255)
}

There is a callback in SchemaDirectiveWiring:

```
default GraphQLInputObjectField onInputObjectField(SchemaDirectiveWiringEnvironment<GraphQLInputObjectField> environment) {
        return environment.getElement();
    }
```

but there is no example on how to use this method to validate the input. There are only [examples](https://github.com/graphql-java-kickstart/samples/tree/master/directives/src/main/java/directives) for [onArgument](https://github.com/graphql-java-kickstart/samples/blob/master/directives/src/main/java/directives/RangeDirective.java) and [onField](https://github.com/graphql-java-kickstart/samples/blob/master/directives/src/main/java/directives/UppercaseDirective.java) callbacks.

Is it possible to provide examples for this kind of validation?

Thanks

**Comment by oliemansm** [comment]
Hi,

I would think implementation of the `onInputObjectField` callback would be similar to the samples of `onArgument` and `onField`. Isn't it possible to deduce it from those samples?

**Comment by PhilippS93** [comment]
Hi @oliemansm . I have tried to use the onField example but I am getting the error:
`Caused by: graphql.AssertException: An output field must be in context to call this method` on calling 

> DataFetcher originalFetcher = environment.getFieldDataFetcher();

**Comment by sumitbadaya27** [comment]
did you get this resolved anyhow? @PhilippS93

**Comment by mohammali** [comment]
Any update?

**Comment by PhilippS93** [comment]
@sumitbadaya27 , no the provided examples are unfortunately not working. @oliemansm , any ideas?

**Comment by sumitbadaya27** [comment]
Yeah not working for me too.
We are using kickstart with java extended validations library.

The library works but then resolvers dont.
And if resolvers work then validation wont.

**Comment by oliemansm** [comment]
I’ll try to take a look at it this weekend.

**Comment by sumitbadaya27** [comment]
Thanks
In case you'd need the link of library for validations, Its here:
https://github.com/graphql-java/graphql-java-extended-validation

**Comment by oliemansm** [comment]
@sumitbadaya27 You're issue seems to be different than what @PhilippS93 raised. You're not getting the validators from that library to work with this project apparently. I don't know how you configured it, but they don't provide a starter but just a library. So you have to expose their SchemaDirectiveWiring into a SchemaDirective bean to be actually configured.

@PhilippS93 graphql-java hasn't documented this either and they refer to https://github.com/graphql-java/graphql-java-extended-validation for example implementation. It uses a FieldDefinition and walks the arguments and input types based on that. I'll see if I can get something like that to work in the sample.

**Comment by sumitbadaya27** [comment]
@oliemansm  Is there any way through in your library in which we can pass typeDefinitionRegistry and runtimeWiring in generating schema in schemaParser builder?
basically this thing:         GraphQLSchema graphQLSchema = new SchemaGenerator().makeExecutableSchema(typeDefinitionRegistry, runtimeWiring);



Also, I configured it correctly as I am getting proper validation message but its just that I am not able to add resolvers and my resolvers fail to work if Validation works.

**Comment by oliemansm** [comment]
@sumitbadaya27 Create a separate issue for that please, this particular issue is not about that library or these questions.

**Comment by sumitbadaya27** [comment]
okay

**Comment by oliemansm** [comment]
The problem is that graphql-java-tools wires the directive on objects and its fields without passing along the constructed `GraphQLInputObjectType`. That's what is preventing that extended-validation library to fail too. Because it does [an instanceof check](https://github.com/graphql-java/graphql-java-extended-validation/blob/bef87f7cbfd6d970aba413eaa51c57852470f77f/src/main/java/graphql/validation/util/DirectivesAndTypeWalker.java#L31) in case the argument is a `GraphQLInputObjectType`, while in fact at that point because of the logic in graphql-java-tools, it is a `GraphQLTypeReference`.

By making some quick changes in graphql-java-tools I was able to pass it along and got it working. Needs a bit more work and clean-up though.

**Comment by sumitbadaya27** [comment]
Till the time you cleanup the code, Can you please provide the code that
you tweaked ans tried and the location where i need to put that code?
So that my work also doesnt stop at this as i am blocked coz of this

**Comment by oliemansm** [comment]
I'll push a snapshot release of graphql-java-tools when it's ready to do so and push the updated samples containing the example code then too.

You can do validation on the resolvers themselves in the meantime. That's a valid workaround available to you, so you're actually not blocked by this.
